### ISSUE TITLE
Set allied/enemy groups and visualize via armor

### ISSUE THREAD

**Issue by Maxopoly** [CivClassic/Finale#6]
Running `/ally Hjaltland` (maybe come up with a better command name) will use ProtocolLib to turn the chest plate and pants of all players on the group into green dyed armor and all of those not on the group into red dyed armor for the player. Properly respect if players are not wearing those armor pieces at all and tie usage to a permission, because it's effectively a bypass on listing members or member checking someone. Properly reset armor if its turned off.

**Comment by xFier** [comment]
I feel like this would be better tied into a namelayer overhaul where groups are replaced with some representation of a 'Nation', so a single nation can make a namelayer 'nation' and then define groups within that. It doesn't really make sense that a single group of people need 20+ groups.

**Comment by Maxopoly** [comment]
I agree and a NameLayer overhaul is one of the things on my todo list, see https://github.com/CivClassic/NameLayer/issues/16, https://github.com/CivClassic/NameLayer/issues/18 and  https://github.com/CivClassic/NameLayer/issues/17

I think that's mostly unrelated to this feature though and if we were to ever change NameLayer in such a way, this feature could easily be adjusted

**Comment by KaloHG** [comment]
it'd probably be beneficial to disable a check on a players allied/enemied status if the player is invisible. This should probably also link into JukeAlert in the sense that if an "enemied" or "allied" player hits a snitch they recieve a proper prefix that colors them according to their standing.

edit: 
I wanted to include that implementing this should be done very carefully as to avoid this turning into an HCF styled system, in the sense where you do like, /team join and you can also view other team info. I think it'd also probably be helpful functionality to permit specific enemy teams to be specified instead of just **all** players.
